src/System.Collections.Immutable/src/System.Collections.Immutable.csproj

```diff
@@ -40,6 +40,7 @@
     <Compile Include="System\Collections\Immutable\ImmutableArray_1.Builder.cs" />
     <Compile Include="System\Collections\Immutable\ImmutableArray_1.Enumerator.cs" />
     <Compile Include="System\Collections\Immutable\ImmutableArray_1.cs" />
+    <Compile Include="System\Collections\Immutable\ImmutableArray_1.FullApi.cs" />
```

Comment:
Can you explain what this extra file is meant to do? I see a bunch of code in it, but I also see a few immutable class related additions to your ecma directory which seems like just code to... compile against? But this looks like additional public API. Why do you need all this new API, and why put it in a separate file?